components/EmotionCardContent.js

```diff
@@ -1,10 +1,33 @@
 import styled from "styled-components";
 import { emotionsIcons } from "./EmotionIcons";
 
-export default function EmotionCardContent({ emotion }) {
+export default function EmotionCardContent({ emotion, emotionTypes }) {
+  const matchedEmotionIcon = (() => {
+    // Hole die zugehörige emotionIcon-ID aus emotionTypes
+    const customEmotion = emotionTypes.find(
+      (item) => item.emotionType === emotion.emotionType
+    );
+    if (customEmotion?.emotionIcon) {
+      // Finde das passende Icon in emotionsIcons basierend auf der ID
+      return emotionsIcons.find(
+        (icon) => icon.id === customEmotion.emotionIcon
+      );
+    }
+    // Fallback: Nutze emotionType als Schlüssel, falls keine emotionIcon-ID definiert ist
+    return emotionsIcons.find(
+      (icon) => icon.emotionType === emotion.emotionType
+    );
+  })();
```

Comment:
this code looks a bit too complex for what it is doing. First it would be best if you would find the correct emotion Icon and color outside of this EmotionCardContent component and inside EmotionCard. Secondly, it seems that you need to distinguish between a predefined emotion and a custom created emotion. This should not be the case. Please make sure to create the same "form" of objects when creating new emotions compared to the predefined emotions. If they need some new keys, please update the predefined emotions accordingly. 
In general it should not matter at this point if an emotion was predefined or user created.

Reply:
refactored now!